src/services/documentRegistry.ts

```diff
@@ -309,7 +310,7 @@ export function createDocumentRegistryInternal(useCaseSensitiveFileNames?: boole
 
         if (!entry) {
             // Have never seen this file with these settings.  Create a new source file for it.
-            const sourceFile = createLanguageServiceSourceFile(fileName, scriptSnapshot, sourceFileOptions, version, /*setNodeParents*/ false, scriptKind);
+            const sourceFile = createLanguageServiceSourceFile(fileName, scriptSnapshot, sourceFileOptions, version, /*setNodeParents*/ false, scriptKind, jsDocParsingMode);
             if (externalCache) {
                 externalCache.setDocument(keyWithMode, path, sourceFile);

```

Comment:
`keyWithMode` should probably also consider `jsDocParsingMode`, so each different jsdoc parse mode gets a different LS document registry cache entry (rather than them randomly overwriting/taking priority over one another). `getDocumentRegistryBucketKeyWithMode` probably needs to be adjusted to take a full `CreateSourceFileOptions` object and work from that (assuming it gets a `jsDocParsingMode` member).

Reply:
I was going to say that each registry has its own setting, so it shouldn't matter, but then I noticed `externalCache`, oops. Will fix.

Reply:
You dont need to add jsDocParsingMode to key since externalCache is internal and operates with same jsDocParsingMode. So just a comment in editorServices should suffice and not add to the key

Reply:
Ah, yeah, you're right; it's tied to the ProjectService which has a fixed JSDocParsingMode. I'll revert it.